Update ReleaseChecklist.md post release steps (#14007)
ethereum/solidity: ReleaseChecklist.md

### Post-release
 - [ ] Make sure the documentation for the new release has been published successfully.
       Go to the [documentation status page at ReadTheDocs](https://readthedocs.org/projects/solidity/) and verify that the new version is listed, works and is marked as default.
 - [ ] Remove "still in progress" warning from the [release notes](https://github.com/ethereum/solidity/releases).

>  - [ ] Merge the [blog posts](https://github.com/ethereum/solidity-blog/pulls).

Author: ```suggestion
 - [ ] Merge the [blog posts](https://github.com/ethereum/solidity-blog/pulls) related to the release.
```

 - [ ] Create a commit to increase the version number on ``develop`` in ``CMakeLists.txt`` and add a new skeleton changelog entry.

>  - [ ] Update the release information section on [soliditylang.org](https://github.com/ethereum/solidity-portal](https://github.com/ethereum/solidity-portal/blob/master/index.html#L22).

Reviewer: ```suggestion
 - [ ] Update the release information section [in the source of soliditylang.org](https://github.com/ethereum/solidity-portal/blob/master/index.html).
```

By they way, linking to a specific line is convenient but risky. It could easily move after some changes in the page structure and the link would become misleading without us noticing. I think that just linking to the whole page is clear enough.

>   - [ ] Announce on [Twitter](https://twitter.com/solidity_lang), including links to the release and the blog post.

Author: ```suggestion
 - [ ] Announce on [Twitter](https://twitter.com/solidity_lang), including links to the release and the blog post.
```

>  - [ ] Announce on [Fosstodon]([https://twitter.com/solidity_lang](https://fosstodon.org/@solidity/), including links to the release and the blog post.

Reviewer: ```suggestion
 - [ ] Announce on [Fosstodon](https://fosstodon.org/@solidity/), including links to the release and the blog post.
```

 - [ ] Share the announcement on Reddit in [``/r/ethdev``](https://reddit.com/r/ethdev/), cross-posted to [``/r/ethereum``](https://reddit.com/r/ethereum/).
 - [ ] Share the announcement on the [Solidity forum](https://forum.soliditylang.org) in the ``Announcements`` category.

>  - [ ] Share the announcement on [Project Updates](https://discord.com/channels/420394352083337236/798974456704925696)

Reviewer: Is this a stable link? Will it keep working or break over time?

Also, would be nice to say which Discord it is (I suspect it's the EF one, but can't tell just looking at the link).

Author: It's a draft. I am still deciding whether or not to explicitly call EF in this doc since this lives in a public repo.

>  - [ ] Share the announcement on [Matrix](https://matrix.to/#/#ethereum_solidity:gitter.im)

Reviewer: ```suggestion
 - [ ] Share the announcement on [`#solidity` channel on Matrix](https://matrix.to/#/#ethereum_solidity:gitter.im)
```

>  - [ ]  Share the announcement on [solc-tooling](https://matrix.to/#/#solc-tooling:matrix.org)

Author: ```suggestion
 - [ ] Share the announcement on [solc-tooling](https://matrix.to/#/#solc-tooling:matrix.org)
```

>  - [ ] Share the announcement on [solc-tooling](https://matrix.to/#/#solc-tooling:matrix.org)

Reviewer: ```suggestion
 - [ ] Share the announcement on [`#solc-tooling`](https://matrix.to/#/#solc-tooling:matrix.org)
```
